packages/adblocker-webextension/adblocker.ts

```diff
@@ -175,10 +210,28 @@ export class BlockingContext {
     sender: Runtime.MessageSender,
   ) => Promise<IMessageFromBackground | undefined>;
 
+  private readonly onCommittedHandler:
+    | ((details: WebNavigation.OnCommittedDetailsType) => void)
+    | undefined;
+
   constructor(private readonly browser: Browser, private readonly blocker: WebExtensionBlocker) {
     this.onBeforeRequest = (details) => blocker.onBeforeRequest(browser, details);
     this.onHeadersReceived = (details) => blocker.onHeadersReceived(browser, details);
     this.onRuntimeMessage = (msg, sender) => blocker.onRuntimeMessage(browser, msg, sender);
+
+    if (
+      this.blocker.config.enablePushInjectionsOnNavigationEvents === true &&
+      usePushScriptsInjection()
```

Comment:
`navigator.userAgent` never changes, so the value of that function can be calculated once and saved into some constant, instead of calling it on each navigation.

Reply:
I can introduce a constant, but note that this code here does not run on each navigation (@chrmod same as you commented). The context is very rarely created. In a normal context, you have one adblocker engine connected to one browser, so the code runs once when the adblocker is created.

But let me introduce the constant, as it does not have any drawback.